node/src/chain.rs

```diff
@@ -168,10 +169,21 @@ pub fn create_firehose_networks(
                     "Configuring firehose endpoint";
                     "provider" => &provider.label,
                 );
+                let subgraph_limit = firehose
+                    .rules
+                    .iter()
+                    .find_map(|r| r.limit_for(&config.node));
```

Comment:
I may be missing something but I don't fully understand why all the other serialization parts are exposed but Web3Rule is hidden. Since it's part of the manifest it would make sense to me that it is public like the other types. Since that's not how it was done before I have followed your suggestions for this PR

Reply:
The concern here is about code modularity: who needs to know how these rules are stored and how to deal with edge cases (like `config.node` doesn't match any rule)

The stuff I wrote about changing the default is a case in point: if the details are hidden behind `firehose.limit_for(..)`, then that's the only place that needs to change to change that behavior. If every place that needs to find the rule for `config.node` has to go through `firehose.rules`, then you need to find all those places and fix them up. Admittedly, there's only one place rn where this is used, but it's a general problem in our code base that too many implementation details are exposed which contributes to making the code base brittle.

There's probably more places in how we deal with the config where policy like this is exposed in too much detail, and we should try to hide more of that ...